connectors/src/connectors/slack/bot.ts

```diff
@@ -593,34 +593,62 @@ async function answerMessage(
   if (buildContentFragmentRes.isErr()) {
     return buildSlackMessageError(buildContentFragmentRes);
   }
+
   let conversation: ConversationType | undefined = undefined;
   let userMessage: UserMessageType | undefined = undefined;
+
   if (lastSlackChatBotMessage?.conversationId) {
-    if (buildContentFragmentRes.value) {
-      const contentFragmentRes = await dustAPI.postContentFragment({
+    const ConversationNotFoundError = class extends Error {
+      constructor(message: string) {
+        super(message);
+        this.name = "ConversationNotFoundError";
+        Object.setPrototypeOf(this, new.target.prototype);
+      }
+    };
+
+    try {
+      if (buildContentFragmentRes.value) {
+        const contentFragmentRes = await dustAPI.postContentFragment({
+          conversationId: lastSlackChatBotMessage.conversationId,
+          contentFragment: buildContentFragmentRes.value,
+        });
+        if (contentFragmentRes.isErr()) {
+          if (contentFragmentRes.error.type === "conversation_not_found") {
+            throw new ConversationNotFoundError("Conversation not found");
+          }
+          return buildSlackMessageError(contentFragmentRes);
+        }
+      }
+
+      const messageRes = await dustAPI.postUserMessage({
         conversationId: lastSlackChatBotMessage.conversationId,
-        contentFragment: buildContentFragmentRes.value,
+        message: messageReqBody,
       });
-      if (contentFragmentRes.isErr()) {
-        return buildSlackMessageError(contentFragmentRes);
+      if (messageRes.isErr()) {
+        if (messageRes.error.type === "conversation_not_found") {
+          throw new ConversationNotFoundError("Conversation not found");
+        }
+        return buildSlackMessageError(messageRes);
       }
+      userMessage = messageRes.value;
+
+      const conversationRes = await dustAPI.getConversation({
+        conversationId: lastSlackChatBotMessage.conversationId,
+      });
+      if (conversationRes.isErr()) {
+        if (conversationRes.error.type === "conversation_not_found") {
+          throw new ConversationNotFoundError("Conversation not found");
+        }
+        return buildSlackMessageError(conversationRes);
+      }
+      conversation = conversationRes.value;
+    } catch (ConversationNotFoundError) {
```

Comment:
In TS you can't capture explicit error types. Here we basically swallow all the errors which is definitely not what we want 🙃.